fix(golint): fix golint error

```
"/home/runner/work/harbor-cluster-operator/harbor-cluster-operator":"/github/workspace" 3b3ac6:935ed6ae656041bab10984aa48116bbf
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
controllers/database/utils.go:145:6: `databaseReadyStatus` is unused (deadcode)
func databaseReadyStatus(properties *lcm.Properties) *lcm.CRStatus {
     ^
controllers/cache/labels.go:33:6: `generateLabels` is unused (deadcode)
func generateLabels(component, role string) map[string]string {
     ^
controllers/cache/utils.go:28:2: `redisRoleMaster` is unused (deadcode)
 redisRoleMaster   = "role:master"
 ^
controllers/storage/provision.go:204:11: ineffectual assignment to `err` (ineffassign)
  secret, err = m.generateS3SecretForChartMuseum(labels)
          ^
controllers/image/image.go:98:15: S1025: the argument is already a string, there's no need to use fmt.Sprintf (gosimple)
  imageAddr = fmt.Sprintf("%s", image)
              ^
controllers/storage/minio.go:157:2: S1008: should use 'return <expr>' instead of 'if <expr> { return <bool> }; return <bool>' (gosimple)
 if m.DesiredMinIOCR.Spec.Image != m.CurrentMinIOCR.Spec.Image {
 ^
controllers/cache/labels.go:23:3: S1031: unnecessary nil check around range (gosimple)
  if labels != nil {
  ^
```

--- a/controllers/cache/generate.go
+++ b/controllers/cache/generate.go
@@ -3,6 +3,8 @@ package cache
 import (
 	"fmt"
 
+	"github.com/goharbor/harbor-cluster-operator/controllers/common"
+
 	redisCli "github.com/spotahome/redis-operator/api/redisfailover/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
@@ -66,9 +68,8 @@ func (redis *RedisReconciler) generateRedisCR() (*unstructured.Unstructured, err
 
 //generateRedisSecret returns redis password secret
 func (redis *RedisReconciler) generateRedisSecret() *corev1.Secret {
-	//labels := MergeLabels(redis.Labels, generateLabels(RoleName, redis.HarborCluster.Name))
 
-	passStr := RandomString(8, "a")
+	passStr := common.RandomString(8, "a")
 
 	return &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{

--- a/controllers/cache/labels.go
+++ b/controllers/cache/labels.go
@@ -20,20 +20,9 @@ func MergeLabels(allLabels ...map[string]string) map[string]string {
 	res := map[string]string{}
 
 	for _, labels := range allLabels {
-		if labels != nil {
-			for k, v := range labels {
-				res[k] = v
-			}
+		for k, v := range labels {
+			res[k] = v
 		}
 	}
 	return res
-}
-
-// generateLabels returns labels
-func generateLabels(component, role string) map[string]string {
-	return map[string]string{
-		"app":       AppLabel,
-		"component": component,
-		component:   role,
-	}
 }

--- a/controllers/cache/utils.go
+++ b/controllers/cache/utils.go
@@ -1,11 +1,8 @@
 package cache
 
 import (
-	"bytes"
 	"fmt"
 	"math/rand"
-	"strings"
-	"time"
 
 	goharborv1 "github.com/goharbor/harbor-cluster-operator/api/v1"
 	"github.com/goharbor/harbor-cluster-operator/lcm"
@@ -25,7 +22,6 @@ const (
 
 	RoleName          = "harbor-cluster"
 	RedisSentinelPort = "26379"
-	redisRoleMaster   = "role:master"
 )
 
 // GetRedisName returns the name for redis resources
@@ -35,38 +31,6 @@ func (redis *RedisReconciler) GetRedisName() string {
 
 func generateName(typeName, metaName string) string {
 	return fmt.Sprintf("%s-%s", typeName, metaName)
-}
-
-// Deprecated
-func RandomString(randLength int, randType string) (result string) {
-	var num string = "0123456789"
-	var lower string = "abcdefghijklmnopqrstuvwxyz"
-	var upper string = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
-
-	b := bytes.Buffer{}
-	if strings.Contains(randType, "0") {
-		b.WriteString(num)
-	}
-	if strings.Contains(randType, "a") {
-		b.WriteString(lower)
-	}
-	if strings.Contains(randType, "A") {
-		b.WriteString(upper)
-	}
-	var str = b.String()
-	var strLen = len(str)
-	if strLen == 0 {
-		result = ""
-		return
-	}
-
-	rand.Seed(time.Now().UnixNano())
-	b = bytes.Buffer{}
-	for i := 0; i < randLength; i++ {
-		b.WriteByte(str[rand.Intn(strLen)])
-	}
-	result = b.String()
-	return
 }
 
 // GetRedisPassword is get redis password

--- a/controllers/database/utils.go
+++ b/controllers/database/utils.go
@@ -141,11 +141,3 @@ func databaseUnknownStatus() *lcm.CRStatus {
 	return lcm.New(goharborv1.DatabaseReady).
 		WithStatus(corev1.ConditionUnknown)
 }
-
-func databaseReadyStatus(properties *lcm.Properties) *lcm.CRStatus {
-	return lcm.New(goharborv1.DatabaseReady).
-		WithStatus(corev1.ConditionTrue).
-		WithReason("database already ready").
-		WithMessage("harbor component database secrets are already create.").
-		WithProperties(*properties)
-}

--- a/controllers/image/image.go
+++ b/controllers/image/image.go
@@ -95,7 +95,7 @@ type ImageLocator interface {
 func GetImage(registry *string, image string) string {
 	var imageAddr string
 	if registry == nil {
-		imageAddr = fmt.Sprintf("%s", image)
+		imageAddr = image
 	} else {
 		imageAddr = fmt.Sprintf("%s/%s", *registry, image)
 	}

--- a/controllers/storage/minio.go
+++ b/controllers/storage/minio.go
@@ -3,6 +3,7 @@ package storage
 import (
 	"context"
 	"fmt"
+
 	"github.com/go-logr/logr"
 	goharborv1 "github.com/goharbor/harbor-cluster-operator/api/v1"
 	"github.com/goharbor/harbor-cluster-operator/controllers/k8s"
@@ -154,11 +155,7 @@ func (m *MinIOReconciler) minioInit() error {
 }
 
 func (m *MinIOReconciler) checkMinIOUpdate() bool {
-	if m.DesiredMinIOCR.Spec.Image != m.CurrentMinIOCR.Spec.Image {
-		return true
-	}
-
-	return false
+	return m.DesiredMinIOCR.Spec.Image != m.CurrentMinIOCR.Spec.Image
 }
 
 func (m *MinIOReconciler) checkExternalUpdate() bool {

--- a/controllers/storage/provision.go
+++ b/controllers/storage/provision.go
@@ -201,14 +201,14 @@ func (m *MinIOReconciler) generateSecretForChartMuseum() (secret *corev1.Secret,
 	switch m.HarborCluster.Spec.Storage.Kind {
 	case s3Storage:
 		labels[LabelOfStorageType] = s3Storage
-		secret, err = m.generateS3SecretForChartMuseum(labels)
+		secret = m.generateS3SecretForChartMuseum(labels)
 	default:
 		return secret, fmt.Errorf(NotSupportType)
 	}
 	return secret, nil
 }
 
-func (m *MinIOReconciler) generateS3SecretForChartMuseum(labels map[string]string) (*corev1.Secret, error) {
+func (m *MinIOReconciler) generateS3SecretForChartMuseum(labels map[string]string) *corev1.Secret {
 	return &corev1.Secret{
 		TypeMeta: metav1.TypeMeta{
 			APIVersion: "v1",
@@ -233,7 +233,7 @@ func (m *MinIOReconciler) generateS3SecretForChartMuseum(labels map[string]strin
 			"AMAZON_REGION":         []byte(m.HarborCluster.Spec.Storage.S3.Region),
 			"AMAZON_ENDPOINT":       []byte(m.HarborCluster.Spec.Storage.S3.RegionEndpoint),
 		},
-	}, nil
+	}
 }
 
 func (m *MinIOReconciler) generateAzureSecret(labels map[string]string) (*corev1.Secret, error) {

